apimodels/test_results.go

```diff
@@ -28,52 +54,112 @@ type CedarTestResult struct {
-		LatestExecution: opts.LatestExecution,
-		FailedSample:    opts.FailedSample,
+		TaskID:       opts.TaskID,
+		Execution:    opts.Execution,
+		DisplayTask:  opts.DisplayTask,
+		TestName:     opts.TestName,
+		Statuses:     opts.Statuses,
+		GroupID:      opts.GroupID,
+		SortBy:       opts.SortBy,
+		SortOrderDSC: opts.SortOrderDSC,
+		BaseTaskID:   opts.BaseTaskID,
+		Limit:        opts.Limit,
+		Page:         opts.Page,
 	}
-	data, err := testresults.GetTestResults(ctx, getOpts)
+}
+
+// GetCedarTestResults makes a request to Cedar for a task's test results.
+func GetCedarTestResults(ctx context.Context, opts GetCedarTestResultsOptions) (*CedarTestResults, error) {
+	data, err := testresults.Get(ctx, opts.convert())
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to get test results for from cedar")
+		return nil, errors.Wrap(err, "getting test results from Cedar")
 	}
 
-	testResults := []CedarTestResult{}
-	if opts.TestName != "" {
-		testResult := CedarTestResult{}
-		if err = json.Unmarshal(data, &testResult); err != nil {
-			return nil, errors.Wrap(err, "failed to unmarshal test result for from cedar")
-		}
-		testResults = append(testResults, testResult)
-	} else {
-		if err = json.Unmarshal(data, &testResults); err != nil {
-			return nil, errors.Wrap(err, "failed to unmarshal test results for from cedar")
-		}
+	testResults := &CedarTestResults{}
+	if err = json.Unmarshal(data, testResults); err != nil {
+		return nil, errors.Wrap(err, "unmarshaling test results from Cedar")
 	}
 
 	return testResults, nil
 }
 
+// GetMultiPageCedarTestResults makes a request to Cedar for a task's test
+// results and returns an io.ReadCloser that will continue fetching and reading
+// subsequent pages of test results if paginated.
+func GetMutliPageCedarTestResults(ctx context.Context, opts GetCedarTestResultsOptions) (io.ReadCloser, error) {
+	r, err := testresults.GetWithPaginatedReadCloser(ctx, opts.convert())
+	if err != nil {
+		return nil, errors.Wrap(err, "getting test results from Cedar")
+	}
+
+	return r, nil
+}
+
+// GetCedarTestResultsStats makes a request to Cedar for a task's test results
+// stats. This route ignores filtering, sorting, and pagination query
+// parameters.
+func GetCedarTestResultsStats(ctx context.Context, opts GetCedarTestResultsOptions) (*CedarTestResultsStats, error) {
+	timberOpts := opts.convert()
+	timberOpts.Stats = true
+	data, err := testresults.Get(ctx, opts.convert())
+	if err != nil {
+		return nil, errors.Wrap(err, "getting test results stats from Cedar")
+	}
+
+	stats := &CedarTestResultsStats{}
+	if err = json.Unmarshal(data, stats); err != nil {
+		return nil, errors.Wrap(err, "unmarshaling test results stats from cedar")
+	}
+
+	return stats, nil
+}
+
+// GetCedarTestResultsFailedSample makes a request to Cedar for a task's failed
+// test result sample (the first 10 failed tests). This route ignores
+// filtering, sorting, and pagination query parameters.
```

Comment:
If there are two methods (`GetCedarTestResultsFailedSample` and `GetCedarTestResultsStats`) that ignore certain parameters in `GetCedarTestResultsOptions`, would it be preferable to split the `GetCedarTestResultsOptions` into two structs so that methods don't selectively ignore some parameters?

Reply:
This is more for future proofing, it is possible that in the future we want to get test stats on filtered results or filter the failed test sample.